src/ff/curve_points.rs

```diff
@@ -0,0 +1,272 @@
+use curve25519_dalek::{
+    constants,
+    ristretto::{CompressedRistretto, RistrettoPoint},
+    Scalar,
+};
+use generic_array::GenericArray;
+use hkdf::Hkdf;
+use sha2::Sha256;
+use typenum::U32;
+
+use crate::{
+    error::Error,
+    ff::{ec_prime_field::Fp25519, Field, Serializable},
+    secret_sharing::{Block, SharedValue},
+};
+
+impl Block for CompressedRistretto {
+    type Size = U32;
+}
+
+///ristretto point for curve 25519
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub struct RP25519(<Self as SharedValue>::Storage);
+
+/// using compressed ristretto point, Zero is generator of the curve, i.e. g^0
+impl SharedValue for RP25519 {
+    type Storage = CompressedRistretto;
+    const BITS: u32 = 256;
+    const ZERO: Self = Self(constants::RISTRETTO_BASEPOINT_COMPRESSED);
+}
+
+impl Serializable for RP25519 {
+    type Size = <<RP25519 as SharedValue>::Storage as Block>::Size;
+
+    fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+        let raw = &self.0.as_bytes()[..buf.len()];
+        buf.copy_from_slice(raw);
+    }
+
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Self {
+        RP25519(CompressedRistretto::from_slice(buf).unwrap())
+    }
+}
+
+impl rand::distributions::Distribution<RP25519> for rand::distributions::Standard {
+    fn sample<R: crate::rand::Rng + ?Sized>(&self, rng: &mut R) -> RP25519 {
+        //Fp25519(Scalar::random(rng: &mut R))
+        let mut scalar_bytes = [0u8; 64];
+        rng.fill_bytes(&mut scalar_bytes);
+        RP25519(RistrettoPoint::from_uniform_bytes(&scalar_bytes).compress())
+    }
+}
+
+impl std::ops::Add for RP25519 {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self((self.0.decompress().unwrap() + rhs.0.decompress().unwrap()).compress())
+    }
+}
+
+impl std::ops::AddAssign for RP25519 {
+    #[allow(clippy::assign_op_pattern)]
+    fn add_assign(&mut self, rhs: Self) {
+        *self = *self + rhs;
+    }
+}
+
+impl std::ops::Neg for RP25519 {
+    type Output = Self;
+
+    fn neg(self) -> Self::Output {
+        Self(self.0.decompress().unwrap().neg().compress())
+    }
+}
+
+impl std::ops::Sub for RP25519 {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self((self.0.decompress().unwrap() - rhs.0.decompress().unwrap()).compress())
+    }
+}
+
+impl std::ops::SubAssign for RP25519 {
+    #[allow(clippy::assign_op_pattern)]
+    fn sub_assign(&mut self, rhs: Self) {
+        *self = *self - rhs;
+    }
+}
+
+///Scalar Multiplication
+///<'a, 'b> `std::ops::Mul<&'b"` Fp25519 for &'a
+impl RP25519 {
+
+    /// # Errors
+    /// Propagates errors from decompressing invalid curve point
+    pub fn s_mul(self, rhs: Fp25519) -> Result<RP25519, Error> {
```

Comment:
I would probably go with `impl <'a, 'b> Mul<Rhs = &'b Fp25519> for &'a RP25519`, given in how many places we panic on failing to decompress